onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -303,33 +303,18 @@ void BasicBackend::StartAsyncInference(Ort::KernelContext& context, OVInferReque
           FillInputBlob(std::move(graph_input_blob), batch_slice_idx, std::move(input_name), context, subgraph_context_);
         } else {
           auto tensor = context.GetInput(subgraph_context_.input_names.at(input_name));
-          auto allocator_name = tensor.GetTensorMemoryInfo().GetAllocatorName();
-          ov_tensor_data_t ov_tensor_key;
-          ort_tensor_key_t ort_tensor_key{tensor.GetTensorRawData(), allocator_name};
-          if (const auto& it = ort_ov_tensor_map.find(ort_tensor_key); it != ort_ov_tensor_map.end()) {
-            ov_tensor_key = it->second;
-          } else {
-            // Does this make sense for both types of allocators?
+          ort_tensor_key_t ort_tensor_key{input_name};
+          auto it = ort_ov_tensor_map.find(ort_tensor_key);
+          if (it != ort_ov_tensor_map.end() || it->second.ort_ptr != tensor.GetTensorRawData()) {
```

Comment:
Why are we comparing the value here, if it was not the same initially then how we can be sure that the value shall remain the same.